packages/langium/src/parser/langium-parser-builder.ts

```diff
@@ -167,12 +167,17 @@ function buildAction(ctx: RuleContext, action: Action): Method {
     return () => ctx.parser.action(action.type, action);
 }
 
-function buildCrossReference(ctx: RuleContext, crossRef: CrossReference): Method {
-    const terminal = crossRef.terminal;
+function buildCrossReference(ctx: RuleContext, crossRef: CrossReference, terminal = crossRef.terminal): Method {
     if (!terminal) {
-        const idx = ctx.consume++;
-        const idToken = getToken(ctx, 'ID');
-        return () => ctx.parser.consume(idx, idToken, crossRef);
+        if (!crossRef.type.ref) {
```

Comment:
Currently the type of a cross-reference refers to a `ParserRule`, but that will change once we implement #91. Furthermore, the approach of finding an assignment to a `name` property does not work with custom `NameProvider` implementations.

What do you think of this approach: When a cross-reference token is not given, look for a terminal rule named `ID`. If that does not exist, create a standard ID rule in memory.

Reply:
I would argue trying to reference a `Type` (#91) without specifying a terminal should result in a validation error. Relying on the existence of some `ID` terminal leads to a lot of implicit behavior which is difficult to debug.

Regarding the `NameProvider`: I'd agree, but again, users should specify the terminal, if they want to prevent the heuristic for finding the name terminal to run. In particular, overriding the actual `name` node probably will happen way less than someone using a `QualifiedName` or something similar as a name (instead of an `ID`). Therefore, I'd say to keep the heuristic, which works for the common case.

Reply:
Xtext uses the `ID` convention, and I never found it to be problematic: https://www.eclipse.org/Xtext/documentation/301_grammarlanguage.html#cross-references

If we're concerned about too much implicit behavior, we can do it like Xtext and reject a grammar with implicit cross-reference token if no `ID` terminal is defined. Or make the token always mandatory (see also discussion at #200).

Reply:
After rethinking, I like your approach more than relying on an ID rule.